packages/doc/src/MarkdownCommentStore.ts

```diff
@@ -0,0 +1,148 @@
+import { readFile } from "fs/promises";
+import { EntityDbMetadata } from "joist-codegen";
+import * as path from "path";
+import { remark } from "remark";
+import { CommentStore, EnumData, FieldSourceType } from "./CommentStore";
+import { hashFile } from "./utils";
+
+/**
+ * A Markdown CommentStore with opinionated but loose semantics around
+ * markdown document structure
+ */
+export class MarkdownCommentStore extends CommentStore {
+  private documents = new Map<string, Promise<null | ReturnType<(typeof remark)["parse"]>>>();
+
+  async forField(entity: EntityDbMetadata, fieldName: string, source: FieldSourceType, generated: boolean) {
+    return this.withGenerated(entity.name, generated, await this.findHeadingContent(entity.name, fieldName));
+  }
+
+  async forEntity(entity: EntityDbMetadata, generated: boolean): Promise<string | void> {
+    return this.findRootContent(entity.name);
+  }
+
+  async forEnum(enumField: EnumData, generated: boolean): Promise<string | void> {
+    return this.findRootContent(enumField.name);
+  }
+
+  async forEnumMember(enumField: EnumData, name: string, generated: boolean): Promise<string | void> {
+    return this.withGenerated(enumField.name, generated, await this.findHeadingContent(enumField.name, name));
+  }
+
+  private withGenerated(file: string, generated: boolean, content: string | undefined) {
+    if (generated) return content;
+
+    return content ? `${content}\n\nautomatically inserted from ${file}.md.` : undefined;
+  }
+
+  async hashForEntity(entity: EntityDbMetadata) {
+    return hashFile(path.join(this.config.entitiesDirectory, `${entity.name}.md`));
+  }
+
+  async hashForEnum(enumData: EnumData) {
+    return hashFile(path.join(this.config.entitiesDirectory, `${enumData.name}.md`));
+  }
+
+  private async getDocument(filename: string) {
+    const existing = this.documents.get(filename);
+
+    if (existing !== undefined) return existing;
+
+    const promise = readFile(path.join(this.config.entitiesDirectory, `${filename}.md`), { encoding: "utf-8" })
+      .then((contents) => {
+        return remark.parse(contents);
+      })
+      .catch(() => null);
```

Comment:
Maybe do a `console.warn` here just so that users can know that we skipped their file due to an error?

Reply:
This essentially catches the file not existing cases, which might be quite noisy to log? It's perhaps a badly placed catch as it's really for the readFile, not the remark.parse which I can't imagine throws that often.

Reply:
Ah I see, agreed the file missing shouldn't log...

I guess you're probably right that `remark.parse` doesn't throw very often/if ever given how lenient markdown is... I was thinking of like parsing ASTs and things, where parse errors are common, but, right the worst thing that happens with Markdown is an misinterpretation of your wip content, and not a true parse error.

Makes sense!